pydocstringformatter/run.py

```diff
@@ -98,21 +84,125 @@ def format_file(self, filename: Path) -> bool:
+        filename: Name of the file the tokens are from.
+
+        Returns:
+            A tuple containing [1] the formatted tokens in a list
+            and a boolean indicating if the tokens were changed.
+
+        Raises:
+            RuntimeError:
+                If the formatters are not able to get to a stable
+                result after 3 iterations.
+
+                It reports what formatters are still modifying the tokens.
+        """
+        formatted_tokens: list[tokenize.TokenInfo] = []
+        is_changed = False
+
+        for index, tokeninfo in enumerate(tokens):
+            new_tokeninfo = tokeninfo
+
+            if _utils.is_docstring(new_tokeninfo, tokens[index - 1]):
+                new_tokeninfo, changers = self.apply_formatters(new_tokeninfo)
+                if changers:
+                    # If the list of changers is empty,
+                    # it means no formatter changed the tokens
+                    is_changed = True
+
+                # Try twice more to get a stable result
+                for _ in range(2):
+                    new_tokeninfo, changers = self.apply_formatters(
+                        new_tokeninfo, changers
+                    )
+
+                if changers:
+                    if len(changers) > 1:
+                        msg = "Conflicting formatters: " + ", ".join(changers)
+                    else:
+                        msg = "Formatter: " + changers[0]
+                        msg += "Is not able to make stable changes on:\n"
+                        msg += new_tokeninfo.string + "\n"
+
+                    if len(changers) == 2:
+                        diff = _compare_formatters(
+                            new_tokeninfo,
+                            self.enabled_formatters[changers[0]],
+                            self.enabled_formatters[changers[1]],
+                            str(filename),
+                        )
+
+                        msg += f"\n{diff}"
+
+                    raise RuntimeError(msg)
+
+            formatted_tokens.append(new_tokeninfo)
+
+        return formatted_tokens, is_changed
+
+    @property
+    def enabled_formatters(self) -> dict[str, _formatting.Formatter]:
+        """Returns a dict of the enabled formatters."""
+        if self._enabled_formatters is not None:
+            return self._enabled_formatters
+
+        enabled = {}
+        for formatter in _formatting.FORMATTERS:
+            if (
+                "default" in formatter.style
+                or any(i in formatter.style for i in self.config.style)
+            ) and getattr(self.config, formatter.name):
+                enabled[formatter.name] = formatter
+
+        self._enabled_formatters = enabled
+        return enabled
+
+    def apply_formatters(
+        self, token: tokenize.TokenInfo, allowed_formatters: list[str] | None = None
+    ) -> tuple[tokenize.TokenInfo, list[str]]:
+        """Applies formatters to a token.
+
```

Comment:
probably i should improve this description to include the fact that it keeps track of what formatters actually change the token